java-frontend/src/test/java/org/sonar/java/UCFGJavaVisitorTest.java

```diff
@@ -86,38 +86,222 @@ public void visit_java_file() {
   }
 
   @Test
-  public void assign_to_field() {
-    // fields are ignored (will change with SONARSEC-121)
+  public void field_assign() {
     Expression.Variable arg = variableWithId("arg");
     UCFG expectedUCFG = UCFGBuilder.createUCFGForMethod("A#method(Ljava/lang/String;)Ljava/lang/String;").addMethodParam(arg)
-      .addBasicBlock(newBasicBlock("1").ret(arg, new LocationInFile(FILE_KEY, 1,69,1,80)))
+      .addBasicBlock(newBasicBlock("1")
+          .assignTo(new Expression.FieldAccess(variableWithId("field")), call("__id").withArgs(arg), new LocationInFile(FILE_KEY, 1,51,1,67))
+          .ret(arg, new LocationInFile(FILE_KEY, 1,69,1,80)))
       .build();
     assertCodeToUCfg("class A { String field; String method(String arg) {this.field = arg; return arg;}}", expectedUCFG);
   }
 
+  @Test
+  public void method_call() {
+    Expression.Variable arg = variableWithId("arg");
+    Expression.Variable aux0 = variableWithId("%0");
+    UCFG expectedUCFG = UCFGBuilder.createUCFGForMethod("A#method(Ljava/lang/String;)Ljava/lang/String;").addMethodParam(arg)
+        .addBasicBlock(newBasicBlock("1")
+            .assignTo(aux0,
+                call("A#method(Ljava/lang/String;)Ljava/lang/String;").withArgs(Expression.THIS, arg),
+                new LocationInFile(FILE_KEY, 1,45,1,61))
+            .ret(aux0, new LocationInFile(FILE_KEY, 1,38,1,62)))
+        .build();
+    assertCodeToUCfg("class A { String method(String arg) { return this.method(arg);}}", expectedUCFG);
+  }
+
+  @Test
+  public void field_access_on_this() {
+    Expression.Variable x = UCFGBuilder.variableWithId("x");
```

Comment:
There is a static import for `variableWithId()`, not need to use `UCFGBuilder.variableWithId`.